### DISCUSSION TITLE
How to use the Premium version of LanguageTool Plus?

### DISCUSSION THREAD

**Discussion by ghost** [valentjn/vscode-ltex#18] [Q&A] [answered]

**Comment by valentjn** [comment, marked as answer]
@dctremblay I suppose you mean the Premium version of LT Plus: https://languagetoolplus.com/. I don’t think that’s possible as there is no API access included there. You would need a Developer API account.

**Comment by ghost** [comment]
Aw oki thanks !

**Comment by melsophos** [comment]
In fact, and contrary to what the [documentation](https://valentjn.github.io/ltex/advanced-usage.html#languagetool-http-servers) says, Premium users can get a free API key (with 10000 requests per day) for private use only by making the request to the support. There is no need to have a developer plan. See [this thread](https://forum.languagetool.org/t/premium-cloud-features-in-vscode-extension/7935) on LanguageTools forum.
(This thread is old but pops in Google's result when searching for this, so I thought it would be worth mentioning.)

**Reply by valentjn** [reply]
Nice. Is this documented somewhere apart from a forum thread?

**Reply by melsophos** [reply]
Not really, I found it by chance by writing to them. I think one could guess it from the following line:
"On-premise and user-based plans are available - contact us for details"
on the [page on the API](https://languagetool.org/proofreading-api), but it is really not explicit.

I found another place which describes the procedure: [LT plugin for Obsidian doc](https://github.com/Clemens-E/obsidian-languagetool-plugin#premium-accounts).

**Reply by bobek** [reply]
Excuse my ignorance, I've just subscribed to the LT premium a couple of minutes ago. Maybe something changed on the LT side? There is a dedicated API token page in settings:

![image](https://user-images.githubusercontent.com/26891/204785661-816f62a4-befe-41de-8854-e9b9d99fdea2.png)

Which links to two documents describing Obsidian and LibreOffice integrations:
- https://languagetool.org/insights/post/product-obsidian/
- https://languagetool.org/insights/post/product-libreoffice/

Furthermore, would it make sense to implement a hybrid mode? Where majority of requests would be sent to local server, while time to time, the cloud/premium server would be consulted?

**Reply by Musta-Pollo** [reply]
Thank you all so much for taking the time to share this :heart:

**Comment by SeniorMars** [comment]
This works for me:

```json
"ltex.languageToolHttpServerUri": "https://api.languagetoolplus.com",
"ltex.languageToolOrg.username": "email",
"ltex.languageToolOrg.apiKey": "your_key",
```

**Reply by bobek** [reply]
aah, yes, I have not shared my final config. Thanks for that. Just for a ref, I am using the following for last 4 months, no issues from the LT so far.

```lua
local opts = {
  on_attach = on_attach,
  cmd = { "ltex-ls" },
  filetypes = { "markdown", "text" },
  flags = { debounce_text_changes = 300 },
  settings = {
    ltex = {
      languageToolOrg = {
        username = "<your_email>",
        apiKey = "<your_api_key>",
      },
      languageToolHttpServerUri = "https://api.languagetoolplus.com",
      checkFrequency = "save",
    }
  },
}
require("lvim.lsp.manager").setup("ltex", opts)
```

**Reply by ecruzolivera** [reply]
Hello @bobek, is this still working for you as is?

**Reply by bobek** [reply]
Yes, I've just switched to Lazy in the meantime, so the config looks like this

```  {
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      flags = { debounce_text_changes = 300 },
      servers = {
        ltex = {
          languageToolOrg = {
            username = "<email>",
            apiKey = "<token>",
          },
          languageToolHttpServerUri = "https://api.languagetoolplus.com",
          checkFrequency = "save",
        },
      },
    },
  },
````

**Reply by Tuebel** [reply]
Can confirm that this URI works. However, when checking long documents, I get a:
`SEVERE: LanguageTool failed with HTTP status code 413`
Checking a smaller selection is my current workaround.
